pulsar/producer_partition.go

```diff
@@ -221,14 +221,25 @@ func newPartitionProducer(client *client, topic string, options *ProducerOptions
 	return p, nil
 }
 
-func (p *partitionProducer) grabCnx() error {
-	lr, err := p.client.lookupService.Lookup(p.topic)
+func (p *partitionProducer) lookupTopic(brokerURL, brokerURLTLS string) (*internal.LookupResult, error) {
+	if len(brokerURL) == 0 && len(brokerURLTLS) == 0 {
```

Comment:
can we move this logic to `lookupService.GetBrokerAddress` to reuse? (Then maybe we need to pass additionally pass topicName to GetBrokerAddress)

Reply:
Done!